grype/pkg/syft_json_provider.go

```diff
@@ -0,0 +1,213 @@
+package pkg
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/anchore/grype/grype/cpe"
+	"github.com/anchore/syft/syft/distro"
+	"github.com/anchore/syft/syft/pkg"
+	syftJson "github.com/anchore/syft/syft/presenter/json"
+	"github.com/anchore/syft/syft/source"
+)
+
+func syftJSONProvider(config providerConfig) ([]Package, Context, error) {
+	var reader io.Reader
+	if config.reader != nil {
+		// the caller is explicitly hinting to use the given reader as input
+		reader = config.reader
+	} else {
+		// try and get a reader from the description
+		if strings.HasPrefix(config.userInput, "sbom:") {
+			// the user has explicitly hinted this is an sbom, if there is an issue return the error
+			filepath := strings.TrimPrefix(config.userInput, "sbom:")
+			sbomReader, err := os.Open(filepath)
+			if err != nil {
+				return nil, Context{}, fmt.Errorf("user hinted 'sbom:' but could read SBOM file: %w", err)
```

Comment:
Should this be "but **_couldn't_** read ..."?

Reply:
lol, woops 👍